scripts/build_test.py

```diff
@@ -14,29 +14,270 @@
+                for line in js_file:
+                    self.assertIn(line, third_party_js_stream.getvalue())
+            counter += 1
+
+    def test_minify_and_create_sourcemap(self):
+        """Tests _minify_and_create_sourcemap with an invalid filepath."""
+        with self.assertRaises(subprocess.CalledProcessError) as calledProcess:
+            build._minify_and_create_sourcemap(
+                'invalid/path/to/input.js', 'invalid/path/to/output.js')
+        calledProcessException = calledProcess.exception
+        # returncode is the exit status of the child process.
+        self.assertEqual(calledProcessException.returncode, 1)
+
+    def test_generate_copy_tasks_for_fonts(self):
+        """Test _generate_copy_tasks_for_fonts to ensure that the number of copy
+        tasks matches the number of font files.
+        """
+        copy_tasks = collections.deque()
+        # Get all filepaths from manifest.json.
+        dependency_filepaths = build.get_dependencies_filepaths()
+        # Setup a sandbox folder for copying fonts.
+        target_fonts_dir = os.path.join('target', 'fonts', '')
+
+        self.assertEqual(len(copy_tasks), 0)
+        copy_tasks += build._generate_copy_tasks_for_fonts(
+            dependency_filepaths['fonts'], target_fonts_dir)
+        # Asserting the same number of copy tasks and number of font files.
+        self.assertEqual(len(copy_tasks), len(dependency_filepaths['fonts']))
+
     def test_insert_hash(self):
-        # pylint: disable=protected-access
-        self.assertEquals(build._insert_hash('file.js', '123456'),
-                          'file.123456.js')
-        self.assertEquals(build._insert_hash('path/to/file.js', '654321'),
-                          'path/to/file.654321.js')
-        self.assertEquals(build._insert_hash('file.min.js', 'abcdef'),
-                          'file.min.abcdef.js')
-        self.assertEquals(build._insert_hash('path/to/file.min.js', 'fedcba'),
-                          'path/to/file.min.fedcba.js')
-        # pylint: enable=protected-access
+        """Test _insert_hash to return correct filenames with provided hashes.
+        """
+        self.assertEqual(
+            build._insert_hash('file.js', '123456'), 'file.123456.js')
+        self.assertEqual(
+            build._insert_hash(
+                'path/to/file.js', '654321'), 'path/to/file.654321.js')
+        self.assertEqual(
+            build._insert_hash('file.min.js', 'abcdef'), 'file.min.abcdef.js')
+        self.assertEqual(
+            build._insert_hash(
+                'path/to/file.min.js', 'fedcba'), 'path/to/file.min.fedcba.js')
+
+    def test_ensure_files_exist(self):
+        """Test _ensure_files_exist raises exception with a non-existent
+        filepath.
+        """
+        random_filepaths = [
+            os.path.join(build.THIRD_PARTY_GENERATED_DEV_DIR, 'random1.js')]
+        with self.assertRaises(OSError) as fileNotExist:
```

Comment:
You can use self.assertRaisesRegexp directly. (I think the codebase has examples.)

Reply:
Done here and elsewhere!